src/Bonsai.ML.Torch/InitializeTorchDevice.cs

```diff
@@ -32,18 +32,30 @@ public class InitializeTorchDevice
         /// <returns></returns>
         public IObservable<Device> Process()
         {
-            InitializeDeviceType(DeviceType);
-            return Observable.Return(new Device(DeviceType, DeviceIndex));
+            return Observable.Defer(() =>
+            {
+                var deviceType = DeviceType;
+                var deviceIndex = DeviceIndex;
+                InitializeDeviceType(deviceType);
+                return Observable.Return(new Device(deviceType, deviceIndex));
+            });
         }
 
         /// <summary>
         /// Initializes the Torch device when the input sequence produces an element.
         /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="source"></param>
         /// <returns></returns>
```

Comment:
```suggestion
        /// <typeparam name="T">The type of the elements in the <paramref name="source"/> sequence.</typeparam>
        /// <param name="source">The sequence containing the notifications used to initialize new Torch devices.</param>
        /// <returns>The sequence of created <see cref="Device"/> objects.</returns>
```

This is the template wording I've been using for documenting type parameters, arguments and return values for initialization / constructor operators.

Reply:
To be resolved later as part of https://github.com/bonsai-rx/machinelearning/issues/54